### PR TITLE
[Hotfix] Update provider registration actions

### PR DESCRIPTION
**Pull request by leb2dg**
### Changes
- Add action for provider registrations
- Add 'need to contact' and 'awaiting response' statuses
- Add admin notes to provider registrations
- Allow longer submissions
- Sort by OAI and display in list
- Update submission to 'accepted' when push user is created

### CODE REVIEW

**Comment by leb2dg** [comment]
@chrisseto, when you get a chance could you review this 🛩

File: share/admin.py [hunk 1]
```diff
@@ -181,13 +183,37 @@ class AccessTokenAdmin(admin.ModelAdmin):
...
+            long_title = str(submission.source_name)
+            home_page = str(submission.source_base_url)
+
+            try:
+                get_user_model().objects.create_trusted_user(username, long_title=long_title, home_page=home_page)
+            except AssertionError:
+                messages.error(request, '{} already exists.'.format(username))
+                continue
+            else:
+                submission.status = 1
```
Comment (chrisseto) [inline share/admin.py]: Prefer `ProviderRegistration.STATUS.accepted` (I think it works)

File: share/admin.py [hunk 2]
```diff
@@ -181,13 +183,37 @@ class AccessTokenAdmin(admin.ModelAdmin):
...
         return ProviderRegistration.STATUS[obj.status].title()
 
+    def create_push_account(self, request, queryset):
+        for submission in queryset:
+            if not submission.direct_source:
+                messages.error(request, '{} is not a direct source.'.format(submission.source_name))
+                continue
+
+            source_url = re.compile('(\.|://)').split(str(submission.source_base_url))
+            username = source_url[-1] + '.' + source_url[-3]
```
Comment (chrisseto) [inline share/admin.py]: I'm not quite sure what the best way to handle this is, but some urls end up looking like. `www.foo.bar`
Maybe just strip out `ww*`s? I've seen a couple `www2`s.

File: share/admin.py [hunk 3]
```diff
@@ -181,13 +183,37 @@ class AccessTokenAdmin(admin.ModelAdmin):
...
     def status_(self, obj):
         return ProviderRegistration.STATUS[obj.status].title()
 
+    def create_push_account(self, request, queryset):
+        for submission in queryset:
+            if not submission.direct_source:
+                messages.error(request, '{} is not a direct source.'.format(submission.source_name))
+                continue
+
+            source_url = re.compile('(\.|://)').split(str(submission.source_base_url))
```
Comment (chrisseto) [inline share/admin.py]: Could you convert the bulk of this into a method on the registration object itself and then add a test for it?

**Comment by leb2dg** [comment]
Closing since this is outdated now

**Pull request closed without merging**